### PR TITLE
Separate radiusd load balancer from main AAA process

### PR DESCRIPTION
**Pull request by julsemaan**
# Description

Separates RADIUS accounting, authentication and load balancing into three different processes
# Impacts

RADIUS authentication and proxy
# Delete branch after merge

YES
# NEWS file entries
## New Features
- Authentication and accounting are now in two different RADIUS processes
- Load balancing in an active/active process now uses a dedicated process

### CODE REVIEW

**Comment by extrafu** [comment]
Shouldn't we also have a configuration for accounting? This would prevent bogus setups where accounting information is sent by thousands of spartans from congesting RADIUS authentication and authorization threads.

**Comment by julsemaan** [comment]
I like the idea yes.

**Comment by julsemaan** [comment]
Accounting has been separated as well

File: addons/packages/packetfence.spec [hunk 1]
```diff
@@ -516,9 +516,7 @@ cd $RPM_BUILD_ROOT/usr/local/pf/conf
 #radius sites-enabled symlinks
 #We standardize the way to use site-available/sites-enabled for the RADIUS server
 cd $RPM_BUILD_ROOT/usr/local/pf/raddb/sites-enabled
-ln -s ../sites-available/control-socket control-socket
```
Comment (louismunro) [inline addons/packages/packetfence.spec:531]: why? these are useful.

File: addons/packages/packetfence.spec [hunk 2]
```diff
@@ -516,9 +516,7 @@ cd $RPM_BUILD_ROOT/usr/local/pf/conf
...
 ln -s ../sites-available/default default
```
Comment (louismunro) [inline addons/packages/packetfence.spec:531]: On the other hand this one is unused anyway.

File: addons/packages/packetfence.spec (hunk 1 above)
Comment (julsemaan) [inline addons/packages/packetfence.spec:531]: The different processes cannot share the same socket so it is specified in the individual files

File: addons/packages/packetfence.spec (hunk 2 above)
Comment (julsemaan) [inline addons/packages/packetfence.spec:531]: I'd prefer not to remove it from this PR as it's unrelated to what this does.

**Comment by louismunro** [comment]
Do you want some help finishing this feature?

**Comment by julsemaan** [comment]
Its completed, I just wanted to check if it would work in a real environment and if the benefits would be there.

And they are :)

**Comment by julsemaan** [comment]
Ready for review

**Comment by louismunro** [comment]
Good enough. Merging.

**Pull request merged**